File: Data Structures & Algorithms/three-integer-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        vector<vector<int>> ans;
        sort(nums.begin(), nums.end());

        for (int i = 0; i < nums.size(); i++) {
            if (i > 0 && nums[i] == nums[i - 1]) continue;
            int target = -1 * nums[i];
            int l = i + 1, r = nums.size() - 1;
            int sum;

            while (l < r) {
                sum = nums[l] + nums[r];
                if (sum == target) {
                    ans.push_back({nums[i], nums[l], nums[r]});
                    l++;
                    r--;
                    while (l < r && nums[l] == nums[l - 1]) l++;
                    while (l < r && nums[r] == nums[r + 1]) r--;
                }
                if (sum < target) {
                    l++;
                }
                if (sum > target) {
                    r--;
                }
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/car-fleet/submission-0.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        time_to_reach = []
        for pos, vel in zip(position, speed):
            time_to_reach.append([pos, (target - pos)/ vel])
        
        time_to_reach.sort(key=lambda x: x[0])
        print(time_to_reach)
        stack = []
        ans = 0
        for t in time_to_reach:
            while stack and t[1] >= stack[-1][1]:
                stack.pop()
            stack.append(t)

        return len(stack)